internal/integration/razorpay/checkout_adapter.go

```
@@ -27,12 +30,100 @@ func (a *CheckoutAdapter) CreatePaymentLink(
+
+	if paymentID := req.GatewayPaymentID; paymentID != "" {
+		rawStatus, err := a.Svc.GetPaymentStatus(ctx, paymentID)
+		if err != nil {
+			return nil, err
+		}
+		status, err := integrations.RazorpayPaymentStatus(rawStatus).ToFlexpricePaymentStatus()
+		if err != nil {
+			// "created" and "authorized" are in-flight, not outcomes.
+			a.Svc.logger.Debug(ctx, "razorpay payment status is not an outcome",
+				"razorpay_payment_id", paymentID, "status", rawStatus)
+			return &interfaces.PaymentState{GatewayPaymentID: paymentID}, nil
```

golangci-lint (2.13.2)

[error] error is not nil (line 73) but it returns nil

(nilerr)